app/src/main/java/com/github/se/orator/ui/friends/FriendsScreen.kt

```diff
@@ -288,3 +354,45 @@ fun DeleteFriendButton(friend: UserProfile, userProfileViewModel: UserProfileVie
             tint = Color.Red)
       }
 }
+
+/**
+ * Computes the current streak of a friend based on their last login date and current streak.
+ *
+ * @param lastLoginDateString The last login date as a string in "yyyy-MM-dd" format. Can be null.
+ * @param currentStreak The current streak value.
+ * @return The streak to be displayed: either currentStreak or 0.
+ */
+fun currentFriendStreak(lastLoginDateString: String?, currentStreak: Long): Long {
+  if (!lastLoginDateString.isNullOrEmpty()) {
+    val lastLoginDate = parseDate(lastLoginDateString) ?: return 0L
+    val currentDate = getCurrentDate()
+    val daysDifference = getDaysDifference(lastLoginDate, currentDate)
+    return when (daysDifference) {
+      0L,
+      1L -> currentStreak // Same day or consecutive day login
+      else -> 0L // Streak broken
+    }
+  }
+  return -1L // No last login date recorded
+}
+
+// Helper functions for date handling
```

Comment:
I think that you could have added these functions in a new file to avoid code duplication with the Firestore or you could have made the methods public in the Firestore to be able to use them here

Reply:
Thanks for the suggestion! I create a DateUtils file to manage all date related methods

Reply:
I also added tests to the dateUtils file to make sure it works as expected

Reply:
I was able to raise the line coverage for the new code to 72%.

Reply:
Those are great changes. Good job!